### PR TITLE
feat!: up python to version 3.11, add dependabot and update dependencies

### LINKED ISSUE #318: Up Python to 3.11 + up dependencies
### Is your feature request related to a problem? Please describe.

Better performance and tracebacks + more!

Maybe time for dependabot?

### Describe the solution you'd like

Python backend running smoothly with 3.11 and updated dependencies

### Describe alternatives you've considered

_No response_

### Additional context

https://github.com/tiangolo/fastapi/pull/5587

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
# Description

Please include a summary of the change and which issue is fixed.

- Updated to Python 3.11
- Added Dependabot
- Updated all dependencies, beside meilisearch related ones see #272 

Fixes #318 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: 2
+updates:
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (AndreasPB) [inline .github/dependabot.yml:6]: Daily might be a little extreme, is there a weekly one?

File: backend/pyproject.toml [hunk 2]
```diff
@@ -5,15 +5,15 @@ description = "Streamchaser's API"
...
-psycopg2-binary = "^2.9.3"
+psycopg2-binary = "^2.9.5"
 typer = "^0.4.0"
 tqdm = "^4.62.3"
 uvicorn = "^0.17.4"
-fastapi = "^0.73.0"
+fastapi = "^0.86.0"
 gunicorn = "^20.1.0"
 httpx = { version = "^0.22.0", extras = ["http2"] }
 pytest-asyncio = "^0.18.1"
```
Comment (AndreasPB) [inline backend/pyproject.toml:19]: Daymn, did we only have 2 outdated ones?

**Review by AndreasPB — CHANGES_REQUESTED** [review]
👋

File: .github/dependabot.yml (hunk 1 above)
Comment (nullxDEADBEEF) [reply .github/dependabot.yml:6]: Yeah there is a weekly one :D I will change it to that

File: backend/pyproject.toml (hunk 2 above)
Comment (nullxDEADBEEF) [reply backend/pyproject.toml:19]: There is prob more, I just wanted to get dependabot up and running and do the job for me of checking :P

Comment (AndreasPB) [reply backend/pyproject.toml:19]: Fair enough

File: .github/dependabot.yml (hunk 1 above)
Comment (nullxDEADBEEF) [reply .github/dependabot.yml]: Now changed to check weekly :D

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**